### DISCUSSION TITLE
Fix some PHAR issues (5.11.2 to 5.12.0)

### DISCUSSION THREAD

**Discussion by ktomk** [captainhook-git/captainhook#193] [General]
Care to share what the phar issues were?

I'm fine with bullet points to some hypertext references. @sebastianfeldmann

**Comment by sebastianfeldmann** [comment]
There were some deprecation warnings during PHAR execution. Nothing major, but being stuck on Symfony v3 for the PHAR was only a temporary fix anyway.
Now the PHAR supports the currently actively supported PHP versions 8.0 - 8.2. Since the PHAR is prepackaged it is way harder to support older versions and be up to date at the same time.
If you need support for older versions you can still install the Cap'n via Composer.

**Comment by ktomk** [comment]
Thanks for fast turn-around. @sebastianfeldmann 

> There were some deprecation warnings during PHAR execution.

Guess this is w/ vanilla PHP install on MacOS? In any case I'm also thinking about filtering those in error handling (promoted to Exception Handling) for production CLI use with own PHP tooling (CLI); must admit that I have no idea if _"Symfony v3"_ allows that thought. (and should add: have not yet fixed it and it's open for months.)

// This may sound as asking for a lot, so only interested if (Yes/No style)  this has been a consideration?

**Reply by sebastianfeldmann** [reply]
I would advise activating deprecations during development basically all the time. You could make an effort to disable the deprecations during the Captains execution but since it made sense to upgrade because of other possible issues in the future, now was the time.

Do you have any problems with the upgrade?

**Reply by ktomk** [reply]
Not yet committed the changes to magerun2 (so no problems so far with the upgrade, build is green but uncertain about PHP version baseline _[ref](https://github.com/netz98/n98-magerun2/pull/1076)_, but that's not "your fault").

What you mention about _production_ (/ _development_) this is clear to me so far, I was however caught off-guard in _CI within pipelines_ which is the main reason I'm postponing such updates _there_ [after first fixes](https://github.com/ktomk/pipelines/issues/21#issuecomment-1240941192), as I understand it merely as an issue with the PHP runtime configuration with PHP 8+ on the _production CLI_ runtime (; and I'm quite conservative on such changes - deprecations are warnings, not errors).

This should go without saying that I think it's fine if you target PHP 8+ for runtime, _here_ just asking quick for context while considering said PHP version baseline for magerun2.

Btw., do you have interest in adding captainhook as _[supported tool](https://github.com/bamarni/symfony-console-autocomplete/#supported-tools)_ to [`bamarni/symfony-console-autocomplete`](https://packagist.org/packages/bamarni/symfony-console-autocomplete)? `captainhookphp/captainhook` is perhaps not that fully interactive all the time, but in general any `symfony-console` based tool should be supported out of the box. Just le me know what you think and if the captain is applicable, we're currently updating the build, and a new contender could come in handy to shake the build a bit.

**Reply by sebastianfeldmann** [reply]
The autocomplete thing sounds interesting. I was looking at the documentation and I'm not getting what is meant by "entrypoint".

> However, [quick setup](https://github.com/captainhookphp/captainhook/discussions/193#quick-setup) only works if the entrypoint is named `console`

**Reply by ktomk** [reply]
the utility name, e.g. `composer` for Composer.

/e: `captainhook` would be added and then its in the quick-setup.
